elixir/apps/api/lib/api/client/channel.ex

```diff
@@ -507,6 +552,90 @@ defmodule API.Client.Channel do
     end
   end
 
+  # This message is sent to the client to request a network flow with a gateway that can server given resource.
+  #
+  # `connected_gateway_ids` is used to indicate that the client is already connected to some of the gateways,
+  # so the gateway can be reused by multiplexing the connection.
+  #
+  # `client_payload` is an arbitrary data that the client wants to send to the gateway, typically it will be
+  # either empty, or contain dns hostname of a given resource to connect to (so that it can be resolved on the
+  # gateway).
```

Comment:
Let's optimistically cull this for now from the protocol, we are going to try and implement it p2p directly.

Reply:
I already did but forgot the comment, thanks!